day10/C++/day10AllPermutations.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+* @author : YashMeh
+* @date   : 28/1/2019
+*/
+#include<bits/stdc++.h>
+using namespace std;
+//Using STL
+void permute(string s)
+{
+   sort(s.begin(),s.end());
+   do
+   {
+      cout<<s<<endl;
+   } while (next_permutation(s.begin(),s.end()));
+}
+//Using recursion
+void permute(string smallPart,int n,string firstChar)
+{
+   if(n==1)
+   {
+      cout<<firstChar+smallPart<<endl;
+      return;
+   }
+   for(int i=0;i<n;i++)
+   {
+      permute(smallPart.substr(1),n-1,firstChar+smallPart[0]);
+      rotate(smallPart.begin(),smallPart.begin()+1,smallPart.end());
+   }
+}
+main()
+{
+   string s="1234";
+   string r="";
+   permute("1234");
```

Comment:
```suggestion
cout << "Using STL, the permutations of " << s << " are: " << endl;
permute(s);
cout << "Using recursion, the permutations of " << s << " are: " << endl;
```

This will make it easier to show that you are using two methods to generate permutations (when I first saw the output I was confused why you were getting 48 permutations for your input instead of 24).